test/dataframes.jl

```diff
@@ -4,64 +4,191 @@ module TestDataFrames
 using Test
 using DataFrames
 using DataFramesMeta
+using Statistics
+
+const ≅ = isequal
+
+df = DataFrame(
+    g = [1, 1, 1, 2, 2],
+    i = 1:5, 
+    t = ["a", "b", "c", "c", "e"],
+    y = [:v, :w, :x, :y, :z],
+    c = [:g, :quote, :body, :transform, missing]
+    )
+
+m = [100, 200, 300, 400, 500]
+
+gq = :g
+iq = :i
+tq = :t
+yq = :y
+cq = :c
+
+gr = "g"
+ir = "i"
+tr = "t"
+yr = "y"
+cr = "c"
+
+@testset "@transform" begin
+    @test @transform(df, n = :i).n == df.i
```

Comment:
in several tests I would also check:
1. if the original `df` was not changed
2. if columns got copied (`!==`) and not just aliased (`===`)
3. not only the transformed column but if all other columns are retained

Reply:
Same for other macros

Reply:
> if columns got copied (`!==`) and not just aliased (`===`)

It looks like that is not actually the behavior

```
julia> df = DataFrame(rand(2,2));

julia> @transform(df, n = :x1).n === df.x1
true
```

Presumably this is not actually behavior we want, so this goes in the `limits of` testset. Do you agree?

Reply:
Yes - for sure we want `@tranform` to make a copy like `transform`, as this is the expected behavior (so I would put the current tests in the set of tests that are likely to be broken after the update).

In general we might consider adding `copycols` kwarg to `@transform` but this can be left for later.